scripts/generate-loggregator-certs

```diff
@@ -2,71 +2,63 @@
 
 set -e
 
-if [ "$1" = "" ]; then
-    echo "usage: $0 <cf-credentials-folder> [--recreate-ca]"
-    echo "       $0 no-bbs-ca [--recreate-ca]"
-    exit 1
-fi
+SCRIPTS_FOLDER="$(dirname "$0")"
 
-readlink() {
-  (
-    cd "$(dirname "$1")"
-    echo "$PWD/$(basename "$1")"
-  )
+ROOT_DEPOT_PATH="$1"
+DEPOT_PATH="loggregator-certs"
+CA_NAME="loggregator-ca"
+BBS_CA_NAME="$CA_NAME"
+
+function create_ca {
+  "${SCRIPTS_FOLDER}"/generate_ca_cert --common-name "loggregatorCA" --depot-path "${ROOT_DEPOT_PATH}/${DEPOT_PATH}" --output-file "$CA_NAME"
 }
 
-# Place keys and certificates here
-depot_path=$(readlink ./loggregator-certs)
-mkdir -p "${depot_path}"
-ca_name="loggregator-ca"
-
-if [ "$1" == "no-bbs-ca" ]; then
-    bbs_ca_name="$ca_name"
-else
-    bbs_ca_cert_path="$(find "$1" -name 'cf-diego-ca.crt')"
-    bbs_ca_key_path="$(find "$1" -name 'cf-diego-ca.key')"
-    bbs_ca_name="cf-diego-ca"
-    if [[ ! -f "$bbs_ca_cert_path" || ! -f "$bbs_ca_key_path" ]]; then
-        echo -e "\e[1m\e[33m[ERROR] Please execute generate-cf-diego-certs before running this script.\e[0m"
-        exit 1
-    fi
-    ln -s "${bbs_ca_cert_path}" "${depot_path}"/
-    ln -s "${bbs_ca_key_path}" "${depot_path}"/
+if [[ "$1" == "--recreate-ca" || "$1" == "no-bbs-ca" || -z "$1" || ! -d "$1" ]]; then
+  echo -e "\e[1m\e[31m[ERROR] Missing ROOT_DEPOT_PATH parameter.\e[0m"
+  exit 1
+fi
+
+test -d "${ROOT_DEPOT_PATH}/${DEPOT_PATH}" || mkdir -p "${ROOT_DEPOT_PATH}/${DEPOT_PATH}"
+
+if [[ "$2" == "--recreate-ca" ]]; then
+  echo -e "\e[1m\e[33m[INFO] Recreating only the loggregatorCA. To recreate the diegoCA, please execute the script '$SCRIPTS_FOLDER/generate-cf-diego-certs'\e[0m"
+  create_ca
+elif [[ -d "$ROOT_DEPOT_PATH/$DEPOT_PATH" && ! -f "$ROOT_DEPOT_PATH/$DEPOT_PATH/$CA_NAME.crt" ]]; then
+  create_ca
 fi
 
-if [[ "$2" == "--recreate-ca" ]] || [[ -z "$2" && ! -f "$depot_path/$ca_name.crt" ]]; then
-  # CA to distribute to loggregator certs
-  certstrap --depot-path "${depot_path}" init --passphrase '' --common-name loggregatorCA
-  mv -f "${depot_path}/loggregatorCA.crt" "${depot_path}/$ca_name.crt"
-  mv -f "${depot_path}/loggregatorCA.key" "${depot_path}/$ca_name.key"
-  mv -f "${depot_path}/loggregatorCA.crl" "${depot_path}/$ca_name.crl"
-elif [[ -z "$2" && -f "$depot_path/$ca_name.crt" ]]; then
-  echo -e "\e[1m\e[33m[INFO] Using existing loggregator CA, recreating only client/server certificates\e[0m"
+if [[ -z "$3" || "$3" != "--no-bbs-ca" ]]; then
+  bbs_ca_cert_path="$(find "$ROOT_DEPOT_PATH" -name 'cf-diego-ca.crt')"
+  bbs_ca_key_path="$(find "$ROOT_DEPOT_PATH" -name 'cf-diego-ca.key')"
+  if [[ -z "$bbs_ca_cert_path" || -z "$bbs_ca_key_path" ]]; then
+    echo -e "\e[1m\e[33m[ERROR] Please execute '$SCRIPTS_FOLDER/generate-cf-diego-certs' before running this script.\e[0m"
+    exit 1
+  fi
+  bbs_ca_cert_path="$(realpath "$bbs_ca_cert_path")"
+  bbs_ca_key_path="$(realpath "$bbs_ca_key_path")"
```

Comment:
`realpath` isn't present on mac systems. Can we use a generic function that works on multiple platforms for this?